File: Whatsapp/OpenWA/sdk/php/src/Resources/CallsResource.php
<?php

declare(strict_types=1);

namespace OpenWA\Resources;

use OpenWA\Http\HttpExecutor;

/**
 * Calls resource — incoming voice/video call handling.
 *
 * Backed by src/modules/call/call.controller.ts.
 */
class CallsResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /**
     * Reject a ringing incoming call (the callId from the call.received event).
     * 404 when the call is not found or no longer ringing. Requires an
     * OPERATOR-level key.
     *
     * @return array<string,mixed>
     */
    public function rejectCall(string $sessionId, string $callId): array
    {
        return $this->http->request('POST', "/api/sessions/{$this->http->encodeSegment($sessionId)}/calls/{$this->http->encodeSegment($callId)}/reject");
    }
}

File: Whatsapp/OpenWA/sdk/java/src/main/java/com/rmyndharis/openwa/resources/CallsResource.java
package com.rmyndharis.openwa.resources;

import static com.rmyndharis.openwa.http.Http.encodeSegment;

import com.rmyndharis.openwa.OpenWAClient;
import com.rmyndharis.openwa.http.HttpMethod;
import com.rmyndharis.openwa.model.SuccessResult;

/** Calls resource — incoming call handling. */
public final class CallsResource {
    private final OpenWAClient client;

    public CallsResource(OpenWAClient client) {
        this.client = client;
    }

    /**
     * Reject a ringing incoming call. The {@code callId} comes from a {@code call.received}
     * webhook event; 404 when the call is not found or no longer ringing.
     */
    public SuccessResult rejectCall(String sessionId, String callId) {
        return client.request(
            HttpMethod.POST,
            "/api/sessions/"
                + encodeSegment(sessionId)
                + "/calls/"
                + encodeSegment(callId)
                + "/reject",
            null,
            null,
            SuccessResult.class);
    }
}
